qiskit-superstaq/qiskit_superstaq/superstaq_job.py

```diff
@@ -12,57 +12,65 @@
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
 
-import time
 from typing import Any, Dict, List, Optional
 
+import general_superstaq as gss
 import qiskit
 
 import qiskit_superstaq as qss
 
 
 class SuperstaQJob(qiskit.providers.JobV1):  # pylint: disable=missing-class-docstring
+
+    TERMINAL_STATES = ("Done", "Canceled", "Failed", "Error")
```

Comment:
It's present in `css.job` as a possible terminal state, and in `superstaq.job` as a `JobStatus`. And so I kept it in case I do something with it since we raise a `SuperstaQException` when a job has been canceled, deleted, or failed. Will remove it now since I don't invoke it.

Reply:
so i just noticed the actual failure status returned by superstaq is "Failed" instead of "Error" - so i think we actually do need this, and every instance of "Error" should also be changed to "Failed" (definitely double-check this though)

Reply:
Wasn't sure of this myself so I initially kept both "Failed" and "Error", but looking at the Superstaq mapping `qiskit.providers.JobStatus.CANCELLED` and `qiskit.providers.JobStatus.ERROR` are set to a "Failed" Superstaq status, so I presume that's the standard in Superstaq. Though does that mean Superstaq also never returns "Canceled"? @vtomole, can you comment on this?

Reply:
> Though does that mean Superstaq also never returns "Canceled"?

That's right. We currently don't give users the ability to cancel jobs.

Reply:
(that said imo there's no reason to remove it as a possibility here - we'll likely support this in the future)

Reply:
Sure, changed "Error" to "Failed" but kept "Canceled" for now in dcf6cca5a4f356abe8ec0610e19e16187cffa570